### ISSUE TITLE
Delta does not color whole lines

### ISSUE THREAD

**Issue by matejcik** [jesseduffield/lazygit#1386]
**Describe the bug**
when using `delta` as pager, the background color for changed lines only goes up to EOL

**To Reproduce**
follow instructions to set up delta: https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Pagers.md

**Problem behavior**

![image](https://user-images.githubusercontent.com/596073/125782634-2e7a0020-87ac-469a-bb69-fb2df887ec0e.png)

**Expected behavior**
this is how it should be, and what delta does when run as `git diff somefile`
![image](https://user-images.githubusercontent.com/596073/125782496-068e72ad-4bcc-451a-aa76-b2733bc5ffe2.png)
it's also what the guide shows:
![image](https://camo.githubusercontent.com/9c18a8359cfafd2e3d8a62da9273608caae044b56bbbda07203c8c0e251df9c4/68747470733a2f2f692e696d6775722e636f6d2f514a70516b46332e706e67)

**Desktop (please complete the following information):**
 - OS: Linux Mint MATE, Mate terminal
 - Lazygit Version: 53ea7df655a8d4893fa623a65d16c7bffded779b

**Comment by matejcik** [comment]
presumably you either need to pass `--width` to delta, or somehow pass the "terminal" width.

I can try to implement the former, either by something like auto-figuring-out options for `delta`, or some sort of configurable parameter on the command line?

**Comment by matejcik** [comment]
this is especially a problem for empty-line differences :eyes:

**Comment by jesseduffield** [comment]
as discussed on slack the file to look at is `vendor/github.com/jesseduffield/gocui/escape.go`: we're failing to properly interpret an escape sequence that says 'finish this line off'

**Comment by matejcik** [comment]
This is fixed in master of gocui already.

I have a 6-line patch for the version in awesome branch, so I can send that as a PR if you like.
Porting the version from master, or making lazygit use gocui master, is probably beyond me though.

**Comment by jesseduffield** [comment]
@matejcik that PR would be good, lemme know if you need any pointers

**Comment by matejcik** [comment]
i believe this was fixed in #1527 ?

**Comment by kkonevets** [comment]
Still not fixed in 0.43.1

```
commit=71ad3fac63a3ef3326021837b49e9497d332818b, build date=2024-07-13T10:24:19Z, build source=binaryRelease, version=0.43.1, os=linux, arch=amd64, git version=2.34.1
```

**Comment by stefanhaller** [comment]
Ah damn. It's actually not "Still not fixed"; it _was_ fixed for two years, but I broke it again in 0.42 with [this commit](https://github.com/jesseduffield/gocui/commit/43d5feee69a197fbeb171ccb3eeff41b4f529143).

Unfortunately, just reverting that commit is not enough to fix the regression; this would expose other bugs that I fixed with that commit, so those then need to be fixed in other ways. I probably won't have time to work on this today, but should be able to get something working next week.
